app/views/questionnaire.py

```diff
@@ -497,40 +499,52 @@ def _redirect_to_location(collection_id, eq_id, form_type, location):
                             group_instance=location.group_instance, block_id=location.block_id))
 
 
-def _get_context(full_routing_path, block, current_location, answer_store):
+def _get_context(full_routing_path, block, current_location, post_form):
+    metadata = get_metadata(current_user)
+    answer_store = get_answer_store(current_user)
+
+    schema_context = _get_schema_context(full_routing_path, current_location, metadata, answer_store)
+
+    block = renderer.render(block, **schema_context)
+
+    variables = g.schema_json.get('variables')
+    if variables:
+        variables = renderer.render(variables, **schema_context)
+
+    if post_form:
+        return {'form': post_form, 'block': block, 'variables': variables}
```

Comment:
I think the we should return from the same place for `post_form` or `form` as the only difference would be the form object generated. As it stands it's not clear they have the same format return data.